src/core/polyhedral/codegen_llvm.cc

```diff
@@ -355,16 +355,34 @@ class CodeGen_TC : public Halide::Internal::CodeGen_X86 {
   }
 };
 
+template <typename CI, typename CO, typename Pred>
+bool all(const CI& inputs, const CO& outputs, Pred p) {
+  for (const auto& v : inputs) {
+    if (not p(v)) {
```

Comment:
this feels like python. is it possible to make a more explicit check here? like you are checking for nullptr etc.?

Reply:
There is no explicit check here. Each instantiation can potentially check for a different thing. 

Also I don't know whether it looks like python but a template predicates exist in every corner of the STL. I'd argue that, by definition, STL's style is idiomatic C++ :stuck_out_tongue:. And actually there's `std::all_of` which I should have used.